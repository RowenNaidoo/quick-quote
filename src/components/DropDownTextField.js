import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

import TextFormField from './TextFormField'
import DropDownFormField from './DropDownFormField'

const DropDownTextField = ({ value, fieldId, headerText, onFieldChanged, errorMessage, options, isRequired }) => {
  const onDropDownChanged = (id, val) => {
    onFieldChanged(fieldId, { countryCode: val, contactNumber: value.contactNumber })
  }

  const onTextChanged = (id, val) => {
    onFieldChanged(fieldId, { countryCode: value.countryCode, contactNumber: val })
  }

  return (<Conatiner>
    <Element>
      {headerText}{isRequired && <RequiredIndicator>*</RequiredIndicator>}
    </Element>
    <Element>
      <DropDownContainer>
        <DropDownFormField
          fieldId={`${fieldId}_dropDown`}
          onFieldChanged={onDropDownChanged}
          options={options}
          selectedValue={value.countryCode}
          rightRadius={false} />
      </DropDownContainer>
      <TextContainer>
        <TextFormField
          value={value.contactNumber}
          fieldId={`${fieldId}_text`}
          onFieldChanged={onTextChanged}
          leftRadius={false} />
      </TextContainer>
    </Element>
    <Element>
      <ValidationMessage>
        {errorMessage}
      </ValidationMessage>
    </Element>
  </Conatiner>)
}

const Conatiner = styled.div`
  display: flex;
  flex-direction: column;
`
const Element = styled.div`
  flex: 1 1 auto;
  justify-content: left;
  display: flex;
  flex-direction: row;
`
const DropDownContainer = styled.div`
  flex: 0 1 100px;
`
const TextContainer = styled.div`
  flex: 1 1 auto;
`
const ValidationMessage = styled.span`
  color: red;
  font-size: 14px;
`
const RequiredIndicator = styled.span`
  color: red;
`

DropDownTextField.propTypes = {
  fieldId: PropTypes.string,
  headerText: PropTypes.string,
  onFieldChanged: PropTypes.func,
  errorMessage: PropTypes.string,
  options: PropTypes.arrayOf(PropTypes.object),
  isRequired: PropTypes.bool
}

export default DropDownTextField