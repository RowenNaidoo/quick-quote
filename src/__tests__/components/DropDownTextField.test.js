import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme'
import DropDownTextField from '../../components/DropDownTextField'


describe('<DropDownTextField />', () => {
  let component
  let mockFieldChanged = jest.fn()

  const defaultProps = {
    value: {contactNumber: '', countryCode: ''},
    fieldId: 'fieldId',
    headerText: 'some header text',
    onFieldChanged: mockFieldChanged,
    errorMessage: '',
    options: [{text: "+61", value: "+61"}],
    isRequired: true
  }

  beforeEach(() => {
    component = shallow(<DropDownTextField {...defaultProps} />)
  })

  test('matches snapshot', () => {
    component = renderer.create(<DropDownTextField {...defaultProps} />)
    expect(component.toJSON()).toMatchSnapshot()
  })

  test('renders dropdown and text field', () => {
    expect(component.find('DropDownFormField')).toHaveLength(1)
    expect(component.find('TextFormField')).toHaveLength(1)
  })
})