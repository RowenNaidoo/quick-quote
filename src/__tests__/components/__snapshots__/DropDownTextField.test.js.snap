// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DropDownTextField /> matches snapshot 1`] = `
<div
  className="sc-fzoLsD iAaohI"
>
  <div
    className="sc-fzpans cqeKxx"
  >
    some header text
    <span
      className="sc-fznZeY ciOsIR"
    >
      *
    </span>
  </div>
  <div
    className="sc-fzpans cqeKxx"
  >
    <div
      className="sc-fzplWN gFyNBt"
    >
      <div
        className="sc-AxgMl bKFcdD"
      >
        <div
          className="sc-AxheI cmsLbY"
        />
        <div
          className="sc-AxheI cmsLbY"
        >
          <select
            className="sc-Axmtr duYEwY"
            id="fieldId_dropDown"
            onChange={[Function]}
            value=""
          >
            <option
              value=""
            >
              Select...
            </option>
            <option
              value="+61"
            >
              +61
            </option>
          </select>
        </div>
        <div
          className="sc-AxheI cmsLbY"
        >
          <span
            className="sc-AxmLO yHMjz"
          />
        </div>
      </div>
    </div>
    <div
      className="sc-fznyAO eqTdrq"
    >
      <div
        className="sc-AxjAm iPShpD"
      >
        <div
          className="sc-AxirZ bqmkbb"
        />
        <div
          className="sc-AxirZ bqmkbb"
        >
          <input
            className="sc-AxiKw jgecmR"
            id="fieldId_text"
            onChange={[Function]}
            type="text"
            value=""
          />
        </div>
        <div
          className="sc-AxirZ bqmkbb"
        >
          <span
            className="sc-AxhCb jNwDnw"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="sc-fzpans cqeKxx"
  >
    <span
      className="sc-fznKkj WVcGc"
    >
      
    </span>
  </div>
</div>
`;
